src/main/java/com/google/api/codegen/gapic/SurfaceGapicProvider.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.gapic;
+
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.codegen.viewmodel.CommonSnippetSetRunner;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.stages.Merged;
+import com.google.api.tools.framework.snippet.Doc;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class SurfaceGapicProvider implements GapicProvider<Interface> {
+  private final Model model;
+  private final CommonSnippetSetRunner snippetSetRunner;
+  private final ModelToViewTransformer modelToViewTransformer;
+
+  private SurfaceGapicProvider(
+      Model model,
+      CommonSnippetSetRunner snippetSetRunner,
+      ModelToViewTransformer modelToViewTransformer) {
+    this.model = model;
+    this.snippetSetRunner = snippetSetRunner;
+    this.modelToViewTransformer = modelToViewTransformer;
+  }
+
+  @Override
+  public List<String> getSnippetFileNames() {
+    return modelToViewTransformer.getTemplateFileNames();
+  }
+
+  @Override
+  public Map<String, Doc> generate() {
+    return generate(null);
+  }
+
+  @Override
+  public Map<String, Doc> generate(String snippetFileName) {
+    // Establish required stage for generation.
+    model.establishStage(Merged.KEY);
+    if (model.getDiagCollector().getErrorCount() > 0) {
+      return null;
+    }
+
+    List<ViewModel> surfaceDocs = modelToViewTransformer.transform(model);
```

Comment:
The following code could be more efficient if it passed `snippetFileName` into the `transform()` method, rather than performing all transformations and then filtering them.